### PR TITLE
test(PromiseObservable): add more tests for fromPromise operator

### PR DESCRIPTION
**Pull request by staltz**
Add more tests to increase test coverage of fromPromise operator (PromiseObservable.create).

### CODE REVIEW

File: spec/observables/from-promise-spec.js [hunk 1]
```diff
@@ -1,21 +1,115 @@
...
-  it('should emit one value from that promise', function (done) {
+  it('should emit one value from a resolved promise', function (done) {
     var promise = Promise.resolve(42);
     Observable.fromPromise(promise)
       .subscribe(function (x) {
         expect(x).toBe(42);
-      }, null,
+      }, done.fail,
       function (x) {
         expect(x).toBe(undefined);
```
Comment (kwonoj) [inline spec/observables/from-promise-spec.js]: Is this required assertion?

File: spec/observables/from-promise-spec.js [hunk 2]
```diff
@@ -1,21 +1,115 @@
...
+  it('should share the underlying promise with multiple subscribers', function (done) {
+    var promise = Promise.resolve(42);
+    var observable = Observable.fromPromise(promise);
+
+    observable
+      .subscribe(function (x) {
+        expect(x).toBe(42);
+      }, done.fail,
+        function (x) {
+          expect(x).toBe(undefined);
```
Comment (kwonoj) [inline spec/observables/from-promise-spec.js]: same as above - is this intention `completed` is being called?

File: spec/observables/from-promise-spec.js (hunk 1 above)
Comment (staltz) [inline spec/observables/from-promise-spec.js]: Not in my opinion.

Comment (kwonoj) [inline spec/observables/from-promise-spec.js]: agreed. same as line 38 and others, I'd like to suggest to remove `undefined` assertion in `completed` along with correcting function signature for `completed`.

Comment (blesh) [inline spec/observables/from-promise-spec.js]: Those are likely leftovers from when `complete` actually did pass a value per the es-observable spec.

Comment (staltz) [inline spec/observables/from-promise-spec.js]: Did that exist? Ew ugly

**Comment by staltz** [comment]
@kwonoj Updated the PR.

File: spec/observables/from-promise-spec.js [hunk 3]
```diff
@@ -32,4 +108,25 @@ describe('Observable.fromPromise', function () {
...
+      expect(reason).toBe('fail');
+      done();
+    });
+
+    Observable.fromPromise(Promise.reject('bad'))
+      .subscribe(
+        done.fail,
+        function (e) {
+          expect(e).toBe('bad');
+          throw 'fail';
```
Comment (kwonoj) [inline spec/observables/from-promise-spec.js]: Seems this test is for `onError` behavior of `subscribe`, does PromiseObservable has special handling for this case?

Comment (staltz) [inline spec/observables/from-promise-spec.js]: @kwonoj Can you elaborate? I don't know what you're saying.

Comment (kwonoj) [inline spec/observables/from-promise-spec.js]: this test (seems) can be applied any kind of observable, i.e 

`Observable.throw('a').subscribe(..., function(e) { throw 'x' }, ...)` 

which emits error. It looks like testing behavior of `Observer::error()`, got curious if `PromiseObservable` has specific differences to handle those.

Comment (staltz) [inline spec/observables/from-promise-spec.js]: Ah, you're correct. I'll try correcting it.

Comment (blesh) [inline spec/observables/from-promise-spec.js]: `fromPromise` is different in that Promises will _eat_ errors thrown in `then` blocks.

**Comment by blesh** [comment]
This LGTM... merged with 2e9246e .. thanks @staltz

**Pull request closed without merging**